### PR TITLE
all: remove support for LLVM 11 and LLVM 12

### PR DESCRIPTION
**Pull request by aykevl**
This removes a lot of backwards compatibility cruft and makes it possible to start using features that need LLVM 13 or newer.
For example:

  * https://github.com/tinygo-org/tinygo/pull/2637
  * https://github.com/tinygo-org/tinygo/pull/2830

### CODE REVIEW

File: compiler/compiler_test.go [hunk 1]
```diff
@@ -28,18 +28,11 @@ type testCase struct {
 func TestCompiler(t *testing.T) {
 	t.Parallel()
 
-	// Check LLVM version.
+	// Determine LLVM version.
 	llvmMajor, err := strconv.Atoi(strings.SplitN(llvm.Version, ".", 2)[0])
```
Comment (deadprogram) [inline compiler/compiler_test.go:32]: Should we fail test if `llvmMajor < 13`?

**Review by deadprogram — APPROVED** [review]

File: src/runtime/gc_conservative.go [hunk 2]
```diff
@@ -233,14 +233,6 @@ func setHeapEnd(newHeapEnd uintptr) {
 // This function can be called again when the heap size increases. The caller is
 // responsible for copying the metadata to the new location.
 func calculateHeapAddresses() {
-	if GOARCH == "wasm" {
-		// This is a workaround for a bug in wasm-ld: wasm-ld doesn't always
-		// align __heap_base and when this memory is shared through an API, it
-		// might result in unaligned memory. For details, see:
-		// https://reviews.llvm.org/D106499
-		// It should be removed once we switch to LLVM 13, where this is fixed.
```
Comment (dgryski) [inline src/runtime/gc_conservative.go:241]: Does this need to be left in place for llvm12 ?

File: src/runtime/gc_globals_conservative.go [hunk 3]
```diff
@@ -4,21 +4,10 @@
...
 // This implementation marks all globals conservatively and assumes it can use
 // linker-defined symbols for the start and end of the .data section.
 func markGlobals() {
-	end := globalsEnd
-	if GOARCH == "wasm" {
-		// This is a workaround for a bug in wasm-ld: wasm-ld doesn't always
-		// align __heap_base and when this memory is shared through an API, it
-		// might result in unaligned memory. For details, see:
-		// https://reviews.llvm.org/D106499
-		// It should be removed once we switch to LLVM 13, where this is fixed.
```
Comment (dgryski) [inline src/runtime/gc_globals_conservative.go:20]: Does this need to be left in place for llvm12 ?

Edit: ignore, thought this was only dropping support for llvm11, not 11 and 12.

**Review by dgryski — APPROVED** [review]
LGTM

File: compiler/compiler_test.go (hunk 1 above)
Comment (aykevl) [reply compiler/compiler_test.go:32]: Why? We're dropping support for LLVM < 13.

**Comment by deadprogram** [comment]
Thanks @aykevl

**Pull request merged**